### PR TITLE
protobuf for TransactionBody

### PR DESCRIPTION
**Pull request by ilblackdragon**

### CODE REVIEW

**Comment by azban** [comment]
this starts the work for https://github.com/nearprotocol/nearcore/issues/306 and enables byte compatibility across languages for transaction bodies.

we only use protobufs for RPC interface for `TransactionBody` for now to unblock client implementations, but still use rust specific `bincode` for anything else going over the network and into storage. the additional work to convert blocks and messages to protobufs will be tracked [here](https://github.com/nearprotocol/nearcore/issues/483)

**Comment by azban** [comment]
check out `transaction/src/lib.rs` for my approach on how we to use protos in our code.. essentially, i just implement `Into` and `From`.. rather than using the `Encode` and `Decode` traits that we have been using previously. from here, we will either use a serde format or directly encode the protobuf into bytes when necessary.

File: node/http/src/types.rs [hunk 1]
```diff
@@ -216,21 +157,23 @@ pub struct ShardBlockResponse {
     // TODO(#301): should have a bs58 format for TransactionResponse
     pub transactions: Vec<SignedTransactionResponse>,
     // TODO(#301): should have a bs58 format for TransactionResponse
+    //#[serde(with = "proto_format")]
```
Comment (ilblackdragon) [inline node/http/src/types.rs]: ?

**Comment by ilblackdragon** [comment]
LGTM for first step.

File: node/http/src/types.rs (hunk 1 above)
Comment (azban) [reply node/http/src/types.rs]: this was a mistake

**Pull request merged**